src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/SessionParameterValueResolver.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\MapSessionParameter;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+final class SessionParameterValueResolver implements ValueResolverInterface
+{
+    public function resolve(Request $request, ArgumentMetadata $argument): array
+    {
+        if (!$attribute = $argument->getAttributesOfType(MapSessionParameter::class, ArgumentMetadata::IS_INSTANCEOF)[0] ?? null) {
+            return [];
+        }
+
+        if (!$request->hasSession()) {
+            return [];
+        }
+
+        if ((!$type = $argument->getType()) || (!class_exists($type) && !interface_exists($type, false))) {
```

Comment:
1. We should dead with nullable type?
2. It does not work with union / intersection type

Reply:
Allowing null allow to use an interface or to create the object only if needed.

I didn't checked for the union type, i'll update the PR

Reply:
compound types aren't compatible with value resolvers IMHO, this one or any other

Reply:
For the record, I tested, an it fails indeed. It tells the class `"App\Foo|AppBar"` does not exist